pkg/test/generic/generic.go

```diff
@@ -65,13 +67,24 @@ func WithAnnotation(key, value string) Option {
 	}
 }
 
-// WithControllerOwnerReference sets the owner reference to the supplied object.
+// WithControllerOwnerReference sets the controller owner reference to the supplied object.
 func WithControllerOwnerReference(owner metav1.Object) Option {
 	return func(meta hivev1.MetaRuntimeObject) {
 		controllerutil.SetControllerReference(owner, meta, scheme.Scheme)
 	}
 }
 
+// WithOwnerReference sets the owner reference to the supplied object.
+// BlockOwnerDeletion is set to true
+func WithOwnerReference(owner hivev1.MetaRuntimeObject) Option {
+	return func(meta hivev1.MetaRuntimeObject) {
+		ownerGVK, _ := apiutil.GVKForObject(owner, scheme.Scheme)
+		ownerRef := metav1.NewControllerRef(owner, ownerGVK)
```

Comment:
```suggestion
		ownerRef := metav1.NewControllerRef(owner, owner.GetObjectKind().GroupVersionKind())
```

If the type meta is not populated for the owner object, then make sure that the builder used to create it adds `WithTypeMeta`.